tactics/eauto.ml

```diff
@@ -373,6 +373,7 @@ let eauto_with_bases ?debug ?depth lems db_list =
   Hints.wrap_hint_warning (e_search_auto ?debug ?depth lems db_list)
 
 let gen_eauto ?debug ?depth lems dbs =
+  Proofview.wrap_exceptions @@ fun () ->
```

Comment:
Rather than using the `wrap_exceptions` function, you should rather `Proofview.Goal.enter`, it is more idiomatic.

Reply:
Is it really?

Reply:
Almost nobody uses wrap_exceptions, except for low-level APIs.

Reply:
It seems to me that a future code-reader would be rather surprised by seeing `Goal.enter`. Won't he question why the `Goal.t` argument is unused and argue that `Goal.enter` might therefore as well be removed?

By using `wrap_exceptions` it is made explicit that the purpose is catching exceptions...